docs/public/specifications/tests/Connectivity-TP/connectivity04.md

```diff
@@ -0,0 +1,235 @@
+failure disabling all of them.
+
+## Scope
+
+It is assumed that *Child Zone* is also tested and reported by [Connectivity01].
+This Test Case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+* "Prefix Database" - The database of IP Prefix data to be used. Possible values
+  are "RIPE" and "Cymru" (the default value).
+* "Cymru Base Name" - If the *Prefix Database* is "Cymru", the default value 
+  is "asnlookup.zonemaster.net".
+* "RIS Whois Server" - If the *Prefix Database* is "RIPE", the default value 
+  is "riswhois.ripe.net".
+
+## Summary
+
+Message Tag                 | Level   | Arguments                   | Message ID for message tag
+:---------------------------|:--------|:----------------------------|:------------------------------------------------------------------------------------------------
+CN04_EMPTY_PREFIX_SET       | ERROR   | ns_ip                       | Prefix database returned no information for IP address {ns_ip}.
+CN04_ERROR_PREFIX_DATABASE  | ERROR   | ns_ip                       | Prefix database error. No data to analyze for IP address {ns_ip}.
+CN04_IPV4_SAME_PREFIX       | WARNING | ns_ip_list, ip_prefix       | The following name server(s) are announced in the same IPv4 prefix ({ip_prefix}): "{ns_ip_list}"
+CN04_IPV4_DIFFERENT_PREFIX  | INFO    | ns_ip_list                  | The following name server(s) are announced in unique IPv4 prefix(es): "{ns_ip_list}"
+CN04_IPV6_SAME_PREFIX       | WARNING | ns_ip_list, ip_prefix       | The following name server(s) are announced in the same IPv6 prefix ({ip_prefix}): "{ns_ip_list}"
+CN04_IPV6_DIFFERENT_PREFIX  | INFO    | ns_ip_list                  | The following name server(s) are announced in unique IPv6 prefix(es): "{ns_ip_list}"
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine Profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [Argument List].
+
+## Test procedure
+
+1. Create the following empty sets:
+   1. Name server IP address ("NS IPv4 IPs")
+   2. Name server IP address ("NS IPv6 IPs")
+   3. IP prefix and name server IP address ("IPv4 Prefix")
+   4. IP prefix and name server IP address ("IPv6 Prefix")
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+3. For each name server IP in *Name Server IP* do:
+   1. Add IPv4 addresses to the *NS IPv4 IPs* set.
+   2. Add IPv6 addresses to the *NS IPv6 IPs* set.
+
+4. For each IP address in *NS IPv4 IPs* and *NS IPv6 IPs* ("NS IP Address"),
+   respectively, do:
+   1. Determine the IP prefix in which *NS IP Address* is announced
+      using *Prefix Database* (details in sections below).
```

Comment:
I do not think it is obvious from this text that you should select either the "Cymru Prefix Lookup" section or "RIPE Prefix Lookup" section below. I will not be a problem for Zonemaster implementation but maybe for others that read the text.